File: webapi/FFischbach.Events.API/Models/Group.cs
namespace FFischbach.Events.API.Models
{
    public class Group
    {
        public int Id { get; set; }
        public required string HashedName { get; set; }
        public required byte[] EncryptedName { get; set; }
        public bool? Approved { get; set; }
        public required string EventId { get; set; }
        public Event? Event { get; set; }
        public int CategoryId { get; set; }
        public Category? Category { get; set; }
        public List<Participant>? Participants { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: webui/src/models/in/Group.ts
import { Event } from "./Event";
import { Participant } from "./Participant";

export class Group {
  constructor(
    public id: number,
    public hashedName: string,
    public name: string,
    public category: string,
    public approved: boolean,
    public contact: Participant,
    public participants: Participant[],
    public createdAt: string,
    public encryptedName?: string,
    public event?: Event
  ) {}
}

export  const clone = (g: Group): Group => {
  const copy = new Group(
    g.id,
    g.hashedName,
    g.name,
    g.category,
    g.approved,
    g.contact,
    g.participants,
    g.createdAt
  );
  return copy;
}
